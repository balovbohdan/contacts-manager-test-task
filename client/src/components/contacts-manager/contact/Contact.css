.main {
    width: 100%;
    height: 70px;

    display: flex;
    align-items: center;

    cursor: pointer;
    position: relative;
    transition: var(--trans);
    background: var(--color-blue);
    box-shadow: 0 2px 3px rgba(0, 0, 0, .2), 0 6px 6px rgba(0, 0, 0, .2);
}

.main:not(:last-of-type) {
    margin-bottom: 20px;
}

.main:hover {
    background-color: rgba(22, 50, 79, .8);
}

.avatar {
    width: 70px;
    height: 100%;

    display: flex;
    align-items: center;
    justify-content: center;

    background-color: rgba(255, 255, 255, .1);
}

.avatar_img {
    width: calc(100% - 20px);
    height: calc(100% - 20px);

    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
}

.name {
    margin-left: 20px;
}

.name_txt {
    font-size: 1.5rem;
    color: var(--color-white);
}

@media (max-width: 450px) {
    .main {
        height: 50px;
    }

    .avatar {
        width: 50px;
    }

    .name_txt {
        font-size: 1rem;
        color: var(--color-white);
    }
}