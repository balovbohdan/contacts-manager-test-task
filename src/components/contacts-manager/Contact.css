.main {
    width: 100%;
    height: 70px;

    display: flex;
    align-items: center;

    cursor: pointer;
    margin-top: 20px;
    position: relative;
    background: var(--color-blue);
    box-shadow: 0 2px 3px rgba(0, 0, 0, .2), 0 6px 6px rgba(0, 0, 0, .2);
}

.avatar {
    width: 70px;
    height: 100%;

    display: flex;
    align-items: center;
    justify-content: center;

    background-color: rgba(255, 255, 255, .1);
}

.avatar_img {
    width: calc(100% - 20px);
    height: calc(100% - 20px);

    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
}

.name {
    margin-left: 20px;
}

.name_txt {
    font-size: 1.5rem;
    color: var(--color-white);
}

.actionsBtn {
    padding: 0 5px;
    cursor: pointer;
    position: absolute;

    right: 10px;
}

.actionsBtn_item {
    width: 5px;
    height: 5px;

    margin: 5px 0;
    border-radius: 50%;

    background-color: var(--color-white);
}

@media (max-width: 450px) {
    .main {
        height: 50px;
    }

    .avatar {
        width: 50px;
    }

    .name_txt {
        font-size: 1rem;
        color: var(--color-white);
    }

    .actionsBtn_item {
        width: 3px;
        height: 3px;
    }
}