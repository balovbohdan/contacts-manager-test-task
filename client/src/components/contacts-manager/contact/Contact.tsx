import * as React from 'react';

import * as T from './types';
import * as css from './Contact.css';

export const Contact = ({contact, toggleContactWindow}:T.Props) => {
    const {id, img, name} = contact;
    const onClick = () => toggleContactWindow({ id });

    return (
        <div onClick={onClick} className={css.main}>
            <Avatar img={img}/>
            <Name name={name}/>
        </div>
    );
};

const Avatar = ({img}:T.AvatarProps) => {
    const style = {
        backgroundImage: `url(${img})`
    };

    return (
        <div className={css.avatar}>
            <div style={style} className={css.avatar_img}/>
        </div>
    );
};

const Name = ({name}:T.NameProps) =>
    <div className={css.name}>
        <p className={css.name_txt}>{name}</p>
    </div>;