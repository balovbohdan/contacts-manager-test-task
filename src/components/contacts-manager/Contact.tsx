import * as React from 'react';

import {T} from '@lib/contacts/contact';

import css from './Contact.css';

export type Props = T.Contact;

type AvatarProps = {
    img:string;
};

type NameProps = {
    name:string;
};

export const Contact = ({img, name}:Props) =>
    <div onClick={alert} className={css.main}>
        <Avatar img={img}/>
        <Name name={name}/>
        <ActionsBtn/>
    </div>;

const Avatar = ({img}:AvatarProps) => {
    const style = {
        backgroundImage: `url(${img})`
    };

    return (
        <div className={css.avatar}>
            <div style={style} className={css.avatar_img}/>
        </div>
    );
};

const Name = ({name}:NameProps) =>
    <div className={css.name}>
        <p className={css.name_txt}>{name}</p>
    </div>;

const ActionsBtn = () =>
    <div className={css.actionsBtn}>
        <div className={css.actionsBtn_item}/>
        <div className={css.actionsBtn_item}/>
        <div className={css.actionsBtn_item}/>
    </div>;