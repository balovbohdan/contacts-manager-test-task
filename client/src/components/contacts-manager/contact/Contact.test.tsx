import * as React from 'react';
import {shallow} from 'enzyme';

import {Contact} from './Contact';

describe('Contact', () => {
    const propsDef = {
        toggleContactWindow: jest.fn(),
        contact: {
            id: 1,
            name: 'name',
            img: '/user.svg',
            phone: '+380971514231'
        }
    };

    it('Should match snapshot', () => {
        const props = { ...propsDef };
        const contact = shallow(<Contact {...props}/>);

        expect(contact).toMatchSnapshot();
    });

    it('Should open contact window after click', () => {
        const props = {
            ...propsDef,
            toggleContactWindow: jest.fn()
        };

        const contact = shallow(<Contact {...props}/>);

        contact.simulate('click');
        expect(props.toggleContactWindow).toBeCalledTimes(1);
    });
});