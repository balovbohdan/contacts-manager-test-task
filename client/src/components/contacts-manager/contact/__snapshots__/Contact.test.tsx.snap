// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Should match snapshot 1`] = `
<div
  className="main"
  onClick={[Function]}
>
  <Avatar
    img="/user.svg"
  />
  <Name
    name="name"
  />
</div>
`;
